tests/NuGet.Services.BasicSearchTests/TestSupport/StartedWebApp.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Serialization;
+using Microsoft.Owin.Hosting;
+using NuGet.Services.BasicSearch;
+
+namespace NuGet.Services.BasicSearchTests.TestSupport
+{
+    public class StartedWebApp : IDisposable
+    {
+        private TestSettings _settings;
+        private INupkgDownloader _nupkgDownloader;
+        private LuceneDirectoryInitializer _luceneDirectoryInitializer;
+        private DataDirectoryInitializer _dataDirectoryInitializer;
+        private PortReserver _portReserver;
+        private IDisposable _webApp;
+
+        public static async Task<StartedWebApp> StartAsync(IEnumerable<PackageVersion> packages = null)
+        {
+            var startedWebApp = new StartedWebApp();
```

Comment:
if you don't want it to be constructed elsewhere, make the constructor private, and the same for initialize async